src/completion/completeStateNames.ts

```diff
@@ -89,13 +92,20 @@ function getCompletionList(
     const list: CompletionList = {
         isIncomplete: false,
         items: items.map(name => {
+            const shouldWrapStateNameInQuotes = languageId === 'asl-yaml' && isStateNameReservedKeyword(name)
```

Comment:
I'd rather do it outside of completeStateNames function. We already have the options to wrap the state name in Quotes. We should keep the reserved words array and any functionality related to it within YAML part of the code.